### ISSUE TITLE
ManualModuleLoader.pas - Error while compiling.

### ISSUE THREAD

**Issue by Dirro** [cheat-engine/cheat-engine#1262]
Steps to reproduce;
0. Download latest source of Cheat Engine
1. Download latest version of Lazarus
2. Say "Ok" on first launch to open to IDE
3. Open cheat engine project in IDE
4. Attempt to build project and get the following error;

`ManualModuleLoader.pas(149,36) Error: Identifier not found "ifthen"`
`ManualModuleLoader.pas(149,66) Fatal: Syntax error, ")" expected but "," found`

-----
I'm using latest updated version of Windows 10 & fresh install. So not sure if I'm missing a package that also needs to be installed or some tweak to the IDE itself for it to compile. *Also using a AMD CPU to do the build.

**Comment by mgrinzPlayer** [comment]
Latest version of Lazarus uses FPC version 3.0.4.
You need FPC3.2.0 or above.

Or just modify line:
```
processhandle:=dbk32functions.OP(ifthen<dword>(GetSystemType<=6,$1f0fff, process_all_access), true, pid);
```
to "if-then statement":
```
  if GetSystemType<=6 then
    processhandle:=dbk32functions.OP(dword($1f0fff), true, pid)
  else
    processhandle:=dbk32functions.OP(dword(process_all_access), true, pid);
```

**Comment by Dirro** [comment]
Thanks a lot for the help! :)

For anyone else who stumbles across this, current pre-built builds are located here for WIN64;
[https://sourceforge.net/projects/lazarus-snapshots/files/Window%2064/](https://sourceforge.net/projects/lazarus-snapshots/files/Window%2064/)

**Comment by mgrinzPlayer** [comment]
FPC3.2.0 beta.

FPC3.2.0 RC1 and above will give you other compile error, just apply patch from here #1248

**Comment by kostic017** [comment]
Why don't you add this info to the Build Instructions section? That and link to cross-compiling addon for 64bit installation?

**Comment by mgrinzPlayer** [comment]
FPC 3.2.0 was released just few days ago (16.06.2020):
https://sourceforge.net/projects/freepascal/files/Win32/3.2.0

I will test this version soon and will edit THIS line.

Currently you have to install Lazarus 2.0.8, install FPC3.2.0 and 64bit addon, launch Lazarus and set path to the new FPC, then rebuild Lazarus 2.0.8. I think it would be better to wait for Lazarus devs to release new Lazarus version built with FPC 3.2.0.

**Comment by mgrinzPlayer** [comment]
Lazarus 2.0.10 build with FPC3.2.0 announced (estimated 4th to 11th July 2020):
https://forum.lazarus.freepascal.org/index.php/topic,50298.0.html
